packages/arb-token-bridge-ui/src/util/withdrawals/helpers.ts

```diff
@@ -54,83 +49,128 @@ export async function mapETHWithdrawalToL2ToL1EventResult(
+    l2TxHash: event.l2TxHash || event.transactionHash,
+    failed
   }
 }
 
+async function isWithdrawalFailed({
+  event,
+  l1Provider,
+  l2Provider,
+  l2ChainId
+}: {
+  event: L2ToL1EventResult & { l2TxHash?: string; transactionHash?: string }
+  l1Provider: Provider
+  l2Provider: Provider
+  l2ChainId: number
+}) {
+  const cacheKey = getL2ToL1MessageCacheKey({
+    event,
+    l2ChainId
+  })
+
+  // cached status is failure
+  const failedMessagesCache = JSON.parse(
+    localStorage.getItem(FAILED_MESSAGES_LOCAL_STORAGE_KEY) || '{}'
+  )
+  if (failedMessagesCache[cacheKey] === true) {
+    return true
+  }
+
+  const messageReader = new L2ToL1MessageReader(l1Provider, event)
+
+  // no cached status for this tx
+  // we check if the transaction has failed
+  if (typeof failedMessagesCache[cacheKey] === 'undefined') {
+    try {
+      await messageReader.getFirstExecutableBlock(l2Provider)
+      saveFailedMessageToCache(cacheKey, false)
+      return false
+    } catch (error) {
+      const err = error as Error & { error: Error }
+      const errorMessage = err && (err.message || err.error?.message)
+
+      if (errorMessage.includes('CALL_EXCEPTION')) {
+        // in classic we simulate `executeTransaction` in `hasExecuted`
+        // which might revert if the L2 to L1 call fail
+        saveFailedMessageToCache(cacheKey, true)
+        return true
+      } else {
+        throw error
+      }
+    }
+  }
+
+  return false
+}
+
+function saveFailedMessageToCache(key: string, failed: boolean) {
+  const currentCache = JSON.parse(
+    localStorage.getItem(FAILED_MESSAGES_LOCAL_STORAGE_KEY) || '{}'
+  )
+  localStorage.setItem(
+    FAILED_MESSAGES_LOCAL_STORAGE_KEY,
+    JSON.stringify({ ...currentCache, [key]: failed })
+  )
+}
+
+function saveExecutedMessageToCache(key: string) {
+  const currentCache = JSON.parse(
+    localStorage.getItem(EXECUTED_MESSAGES_LOCAL_STORAGE_KEY) || '{}'
+  )
+  localStorage.setItem(
+    EXECUTED_MESSAGES_LOCAL_STORAGE_KEY,
+    JSON.stringify({ ...currentCache, [key]: true })
+  )
+}
+
 export async function getOutgoingMessageState(
   event: L2ToL1EventResult,
   l1Provider: Provider,
   l2Provider: Provider,
   l2ChainID: number
 ) {
-  const cacheKey = getExecutedMessagesCacheKey({
+  const cacheKey = getL2ToL1MessageCacheKey({
     event,
     l2ChainId: l2ChainID
   })
 
+  // cached status is executed
   const executedMessagesCache = JSON.parse(
-    localStorage.getItem('arbitrum:bridge:executed-messages') || '{}'
+    localStorage.getItem(EXECUTED_MESSAGES_LOCAL_STORAGE_KEY) || '{}'
   )
   if (executedMessagesCache[cacheKey]) {
     return OutgoingMessageState.EXECUTED
   }
 
   const messageReader = new L2ToL1MessageReader(l1Provider, event)
 
-  try {
-    return await messageReader.status(l2Provider)
-  } catch (error) {
-    return OutgoingMessageState.UNCONFIRMED
-  }
-}
+  const l1Network = await l1Provider.getNetwork()
+  const { isMainnet } = isNetwork(l1Network.chainId)
```

Comment:
perhaps we should pass the chainId into `mapWithdrawalToL2ToL1EventResult` then into this function so that we don't need to use `l1Provider` to get the chain id?